bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -279,13 +279,13 @@ int ping_island(int argc, char * argv[]) {
     if (!strcmp(windowsVersion, WINDOWS7SP1) || !strcmp(windowsVersion, WINDOWS7)) {
         TCHAR windir[MAX_PATH];
         GetWindowsDirectory(windir, MAX_PATH);
-        char* dllPath = "/System32/Ucrtbase.dll";
-        char* absDllPath = malloc(strlen(windir) + strlen(dllPath) + 1);
+        wchar_t* dllPath = L"\\System32\\Ucrtbase.dll";
+        wchar_t* absDllPath = malloc(sizeof(windir) + sizeof(dllPath));
```

Comment:
YOu're allocating too much. You will allocate MAX_PATH + dllPath. But all paths in Windows (nearly) are under MAX_PATH.

Allocate a MAX_PATH variable

Reply:
smart